### ISSUE TITLE
(Bliss Os) longer it gets, the more error that has

### ISSUE THREAD

**Issue by MadanaSnake** [Xtr126/XtMapper#113]
Hey, while i'm using xtmapper with my fps game for a minute or an hour, the aming mode is lagging and has to stop manually (or pause), the cursor not showing up, and the aming mode stuck

> Os : Bliss Os 16 (Android 13)

**Comment by Xtr126** [comment]
> Hey, while i'm using xtmapper with my fps game for a minute or an hour, the aming mode is lagging and has to stop manually (or pause), the cursor not showing up, and the aming mode stuck
> 
> > Os : Bliss Os 16 (Android 13)

Just to make sure, by default Bliss OS comes with an old version of XtMapper, have you tried upgrading it to the latest version?

**Comment by MadanaSnake** [comment]
Yes, i upgrade the xtmapper with the universal one, and the latest one

**Comment by Xtr126** [comment]
Oh it's not that then.
After a discussion with Bliss OS developers we are planning to integrate XtMapper at system-level instead of being a user-app like it is now. It will take time though, but should solve these weird problems. Or may introduce new ones, have to try.
Thanks for reporting this issue.

**Comment by MadanaSnake** [comment]
Ok, thank you. One more thing,
if the xtMapper is in the system-level, do i have to upgrade the bliss os?

**Comment by Xtr126** [comment]
> Ok, thank you. One more thing, if the xtMapper is in the system-level, do i have to upgrade the bliss os?

Yes, to integrate it as a system app, system has to be modified.

**Comment by MadanaSnake** [comment]
And one more thing,
While mapping, the deleted mapping are not saved. But new mapping and moved mapping are saved...

**Comment by Xtr126** [comment]
Fixed! Thanks for pointing out.

**Comment by MadanaSnake** [comment]
i should update the app?
if that so, thanks!

**Comment by MadanaSnake** [comment]
i should update the app?
if that so, thanks!

**Comment by Xtr126** [comment]
You can test the changes by downloading the artifacts from here  https://github.com/Xtr126/XtMapper/actions

**Comment by MadanaSnake** [comment]
wow, the ui is is new!
and the removal button works too!
Thanks!

**Comment by Xtr126** [comment]
Glad you like it :)
